### PR TITLE
Introduce Tencent as a known JVM vendor

### PR DESCRIPTION
**Pull request by johnshajiang**
### Context
Tencent is a notable contributor to OpenJDK, and has been publishing its own OpenJDK distribution, exactly Kona JDK, for years.
- https://github.com/Tencent/TencentKona-8
- https://github.com/Tencent/TencentKona-11
- https://github.com/Tencent/TencentKona-17

And SDKMAN already supports Tencent Kona JDK
- https://sdkman.io/jdks#kona

This PR just adds Tencent as a known JVM vendor so that we can easily apply Kona JDK with Gradle toolchain feature.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by lkasso — APPROVED** [review]
Docs approved.

**Comment by johnshajiang** [comment]
@jvandort 
Could you please review this PR?
If you have any concern, please let me know.

**Comment by jvandort** [comment]
Please rebase this into master and we can get this merged

**Comment by johnshajiang** [comment]
@jvandort 
I just updated the commit.
Could you please review it?

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
Have you tested this? Can you confirm that the `java.vendor` property on these VMs are equal to `tencent`?

Since we do not test provisioning and using these toolchains from all vendors, I will take your word on this one

**Comment by jvandort** [comment]
It looks like your PR is failing sanityCheck. 

Please run `./gradlew sanityCheck` locally and address the issues reported by the binary compatibility check

**Comment by johnshajiang** [comment]
> Have you tested this?

When this PR was submitted, I did run the tests, like `sanityCheck`.
But, I just tried to execute that task and always encountered the below problem,
```
Error resolving plugin [id: 'org.gradle.toolchains.foojay-resolver-convention', version: '0.7.0']
> A problem occurred configuring project ':build-logic-commons:basics'.
   > Dependency verification failed for configuration ':build-logic-commons:basics:classpath'
     One artifact failed verification: kotlin-stdlib-1.9.20.jar (org.jetbrains.kotlin:kotlin-stdlib:1.9.20) from repository Gradle Central Plugin Repository
```

The failed task log contains the below lines,
```
[16:30:00]W:		 [Gradle failure report] Caused by: org.gradle.api.GradleException: Detected binary changes.
[16:30:00]W:		 [Gradle failure report]     - current: gradle-kotlin-dsl-shared-runtime-8.6.jar, gradle-api-metadata-8.6.jar,  ...
[16:30:00]W:		 [Gradle failure report]     - baseline: gradle-antlr-8.5.jar, gradle-api-metadata-8.5.jar, ...
```

The current jars are different from the baseline jars. But I don't understand it.
Does it have something to do with my changes?

**Comment by johnshajiang** [comment]
> Can you confirm that the java.vendor property on these VMs are equal to tencent?

The value of that property is `Tencent`.
I supposed it is case insensitive.
```
KnownJvmVendor(String indicatorString, String displayName) {
    this.indicatorString = indicatorString;
    this.indicatorPattern = Pattern.compile(indicatorString, Pattern.CASE_INSENSITIVE);
    this.displayName = displayName;
}
```

File: platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/JvmVendorSpec.java [hunk 1]
```diff
@@ -83,6 +83,14 @@ public abstract class JvmVendorSpec {
 
     public static final JvmVendorSpec SAP = matching(KnownJvmVendor.SAP);
 
+    /**
+     * A constant for using <a href="https://github.com/Tencent/TencentKona-8">Tencent Kona JDK</a> as the JVM vendor.
+     *
+     * @since 8.5
```
Comment (jvandort) [inline platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/JvmVendorSpec.java:89]: The version of Gradle on master has changed. You need to update this to `8.6`

**Review by jvandort — CHANGES_REQUESTED** [review]

**Comment by jvandort** [comment]
> > Can you confirm that the java.vendor property on these VMs are equal to tencent?
> 
> The value of that property is exactly `Tencent`. I supposed it is case insensitive.
> 
> ```
> KnownJvmVendor(String indicatorString, String displayName) {
>     this.indicatorString = indicatorString;
>     this.indicatorPattern = Pattern.compile(indicatorString, Pattern.CASE_INSENSITIVE);
>     this.displayName = displayName;
> }
> ```

The current value seems fine then

File: platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/JvmVendorSpec.java (hunk 1 above)
Comment (johnshajiang) [reply platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/JvmVendorSpec.java]: Thanks very much!
I just pushed a new commit.

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test and merge

**Comment by johnshajiang** [comment]
@jvandort 
Thanks for approving and merging this PR very much!

In addition, I just moved to another network environment, and resolved all the dependencies and build Gradle successfully.
My project now can use Kona JDK builds via Gradle toolchain with my Gradle build.

**Comment by jvandort** [comment]
Glad you got thr build working. These changes should be present in our nightly soon

**Pull request merged**